### PR TITLE
[AC-7542] Ability to filter /api/v1/user/ by date modified

### PR DESCRIPTION
**Pull request by kaymation**
[JIRA](https://masschallenge.atlassian.net/browse/AC-7542)

Turns out we already have this functionality, so this PR just adds a test to verify the overall functionality requested by the ticket indeed operated correctly 

To Test:
- navigate to `/api/v1/user`
- add `?updated_at.after=2016-04-23T18:25:43.511Z`(or any proper ISO 8601 date)
- * should only report users modified after this date
- replace the parameter with `?updated_at.before=2016-04-23T18:25:43.511Z`(or any proper ISO 8601 date)
- * should only report users modified before this date
- now, try adding `?updated_at.after=2016-04-23T18:25:43.511Z&?updated_at.before=2019-04-23T18:25:43.511Z`(or any two proper iso dates where the latter is more recent than former)
- * should only report users modified between these two dates

### CODE REVIEW

File: web/impact/impact/tests/test_user_list_view.py [hunk 1]
```diff
@@ -507,6 +507,21 @@ def test_updated_at_after_datetime_filter(self):
             assert _contains_user(updated_exactly, response.data)
             assert _contains_user(updated_after, response.data)
 
+    def test_updated_at_filter_when_same_day(self):
```
Comment (jonkiparsky) [inline web/impact/impact/tests/test_user_list_view.py:510]: drive-by comment: there is a lot of duplication in these tests. While we're here, it would be great if we could clean that up and make it pretty.

**Comment by geofrocker** [comment]
will do QA

**Review by geofrocker — COMMENTED** [review]
I am not seeing the parameter names that have been updated. Only seeing a test. I am I missing something? @kaymation 
>so this PR just updates the parameter names with those specified in the ticket

**Review by geofrocker — APPROVED** [review]
Code and QA look good

**Review by philophilo — APPROVED** [review]
Tests look good ✅ 
Date formats make the query ✅
